### PR TITLE
feat(accessibility): support large content viewer on iOS

### PR DESCRIPTION
**Pull request by bacarybruno**
## Summary:

Referring to the [iOS Large Content Viewer](https://developer.apple.com/videos/play/wwdc2019/261/):
iOS Tab Bars can't grow with dynamic text, but the Large Content Viewer helps them to be seen by people with low vision.

Currently on React Native we don't expose the properties that can help implementing iOS [UILargeContentViewerItem](https://developer.apple.com/documentation/uikit/uilargecontentvieweritem) protocol.
The goal of this PR is to expose the necessary props.

In this PR, I'm exposing 2 props:
- `accessibilityShowsLargeContentViewer`: to enable the large content viewer
- `accessibilityLargeContentTitle`: to define the large content viewer title

I plan to use this to open a PR on react-navigation so that bottom tabbars can implement largeContentViewer.
Should fix https://github.com/facebook/react-native/issues/30892

## Changelog:
[IOS] [ADDED] - Support LargeContentViewer on iOS for better accessibility

### CODE REVIEW

**Comment by joevilches** [comment]
Hey @bacarybruno, thanks for taking on an old issue and making RN more accessible! 

At first glance, this needs new arch support. Do you mind adding that? You can take a look at this example for some code pointers: https://github.com/facebook/react-native/commit/d756d4394d0f5596b208cc595079ff0740ba5f82. The big one is RCTViewComponentView, but we also need to parse these props in the new renderer, so some of the headers in that commit should be of use. You can test this by enabling new arch on whatever app you are testing on: https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-apps.md. This might also be of use: https://reactnative.dev/docs/the-new-architecture/landing-page

**Review by joevilches — CHANGES_REQUESTED** [review]
Requesting changes for new arch support

**Comment by bacarybruno** [comment]
I'll try to handle it in the coming days. Thanks for the detailed comment 🙏

File: packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm [hunk 1]
```diff
@@ -346,6 +348,47 @@ - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &
...
+  if (oldViewProps.accessibilityLargeContentImage != newViewProps.accessibilityLargeContentImage) {
+    if (@available(iOS 13.0, *)) {
+      self.largeContentImage = nil;
+
+      NSURL *imageURL = NSURLFromImageSource(newViewProps.accessibilityLargeContentImage);
+      if (!imageURL) {
+        return;
+      }
+
+      UIImage *image = RCTImageFromLocalAssetURL(imageURL);
```
Comment (bacarybruno) [inline packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm:382]: @joevilches I'll need your opinion here.
What would be the best way to get a UIImage from an `ImageSource`?
I can't see anything about that 🙏

Comment (bacarybruno) [reply packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm:382]: I'm still interested in the answer, but actually for this PR I think I'll focus on largeContentTitle.
For the `largeContentImage` there are a lot of things to consider, like:
- Do we want to pass only local images as props, or remote images as well
- Do we want to support svg images, can be useful for apps that already use svgs are tabbar icons
I'll open a separate PR for largeContentImage 🙂

Comment (joevilches) [reply packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm]: @bacarybruno For the image question, I am not sure. You could take a look at RCTImageComponentView.mm which may give you some ideas (I took a look and was not successful :/ ). Splitting the PR seems like a good path forward. I can help find someone more knowledgable than me when your next PR comes out

**Review by joevilches — APPROVED** [review]

File: packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm (hunk 1 above)
Comment (bacarybruno) [reply packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm]: Okay thank you. Please don't hesitate to tell me if there is something left to do for this PR.

Comment (bacarybruno) [reply packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm]: @joevilches the linting and tests failed in the previous commit. I just [pushed an update](https://github.com/facebook/react-native/pull/45903/commits/6dafbab7b8e641240c50402c5a47542af0abd651) for that.

**Pull request closed without merging**